dask/dataframe/groupby.py

```diff
@@ -1245,9 +1245,20 @@ def __init__(
         if any(isinstance(key, pd.Grouper) for key in by_):
             raise NotImplementedError("pd.Grouper is currently not supported by Dask.")
 
+        # slicing key applied to _GroupBy instance
+        self._slice = slice
+
+        # Check if we can project columns
+        projection = None
+        if isinstance(self._slice, (str, list)):
```

Comment:
What other types are valid here?

Reply:
The `slice=` kwarg is set by the key in a `DataFrameGroupBy.__getitem__` call. Therefore, I suppose it can be anything representing a column selection. I opted to keep the projection simple here for now, but we certainly *can* generalize this.

Reply:
Hopefully the latest code supports all (reasonable) `self._slice` types (numpy scalar, str, list, tuple, index-like, and series-like)